File: Semester 4/Medii de proiectare si programare/Laborator/Laborator 11/FestivalMuzicaJavaFXClientServer/rest/src/main/java/client/ShowsClientRunner.java
package client;

import ro.mpp2026.festivalmuzicajavafx.domain.Show;
import ro.mpp2026.festivalmuzicajavafx.domain.ShowFilter;

import java.time.LocalDate;
import java.util.Arrays;

public class ShowsClientRunner {
    public static void main(String[] args) {
        ShowsClient client = new ShowsClient();

        Show newShow = new Show(null, "Test Artist", LocalDate.now().plusDays(2), "Test Hall", 100);
        Long id = client.save(newShow);
        System.out.println("Created show id: " + id);

        Show created = client.findById(id);
        System.out.println("Fetched: " + created);

        created.setRemainingSeats(80);
        client.update(id, created);
        System.out.println("Updated show id: " + id);

        Show[] all = client.findAll();
        System.out.println("All shows: " + Arrays.toString(all));

        ShowFilter filter = ShowFilter.builder()
                .artistName("Test Artist")
                .build();
        Show[] filtered = client.findAllFiltered(filter);
        System.out.println("Filtered shows: " + Arrays.toString(filtered));

        client.delete(id);
        System.out.println("Deleted show id: " + id);
    }
}

File: Semester 4/Medii de proiectare si programare/Laborator/Laborator 10/FestivalMuzicaCSharpWFA/rest/ShowsClientRunner.cs
using FestivalMuzicaCSharp.Domain;
using System.Linq;

namespace Rest;

public static class ShowsClientRunner
{

    public static async Task RunAsync()
    {
        using var client = new ShowsClient();

        var newShow = new Show(0, "Test Artist", DateOnly.FromDateTime(DateTime.Now.AddDays(2)), "Test Hall", 100);
        var id = await client.SaveAsync(newShow);
        Console.WriteLine($"Created show id: {id}");

        var created = await client.FindByIdAsync(id);
        Console.WriteLine($"Fetched: {created}");

        if (created is not null)
        {
            created.RemainingSeats = 80;
            await client.UpdateAsync(id, created);
            Console.WriteLine($"Updated show id: {id}");
        }

        var all = await client.FindAllAsync();
        Console.WriteLine($"All shows: {string.Join(", ", all.Select(show => show.ToString()))}");

        var filter = new ShowFilter("Test Artist", null, null, null);
        var filtered = await client.FindAllFilteredAsync(filter);
        Console.WriteLine($"Filtered shows: {string.Join(", ", filtered.Select(show => show.ToString()))}");

        await client.DeleteAsync(id);
        Console.WriteLine($"Deleted show id: {id}");
    }
}
